File: tumbuhin_flutter/lib/shared/services/alert_service.dart
import '../../../core/api/api_client.dart';
import '../models/alert.dart';

class AlertService {
  final ApiClient _apiClient;

  AlertService(this._apiClient);

  Future<List<SmartAlert>> getAlerts() async {
    final response = await _apiClient.dio.get('/api/v1/alerts');
    return (response.data as List).map((e) => SmartAlert.fromJson(e)).toList();
  }

  Future<void> markAsRead(String id) async {
    await _apiClient.dio.patch('/api/v1/alerts/$id/read');
  }

  Future<void> markAllAsRead() async {
    await _apiClient.dio.post('/api/v1/alerts/read-all');
  }

  Future<Map<String, bool>> getNotificationSettings() async {
    final response = await _apiClient.dio.get('/api/v1/notifications/settings');
    return Map<String, bool>.from(response.data);
  }

  Future<void> updateNotificationSettings(Map<String, bool> settings) async {
    await _apiClient.dio.put('/api/v1/notifications/settings', data: settings);
  }
}

File: web/src/lib/api/alertService.ts
import { apiGet, apiPut, apiPost } from './client';

export const alertService = {
  getAlerts: () => apiGet<any[]>('/api/v1/business-profile/alerts'),
  getUnreadCount: () => apiGet<{ count: number }>('/api/v1/business-profile/alerts/count'),
  markAsRead: (id: string) => apiPut(`/api/v1/business-profile/alerts/${id}/read`, {}),
  markAllAsRead: () => apiPost('/api/v1/business-profile/alerts/read-all', {}),
};
